### DISCUSSION TITLE
How to embed JavaScript in HTML within Wiki.js to support sending a POST HTTP request to the server

### DISCUSSION THREAD

**Discussion by chenzhiqiang0518** [requarks/wiki#7322] [Help / Questions] [answered]
I create new html document, and write demo page ,but when click button, cannot send the post http request.
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>HTTP POST demo</title>
</head>
<body>
    <h1>HTTP POST demo</h1>
    <button id="sendRequestButton">send request</button>

    <script>
    document.getElementById("sendRequestButton").addEventListener("click", function() {
        const url = "http://127.0.0.1:7002/example";  
        const data = {
            "key": "value"
        };

        fetch(url, {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(data)
        })
        .then(response => response.json())
        .then(data => {
            alert("success: " + JSON.stringify(data));
        })
        .catch((error) => {
            console.error("fail:", error);
        });
    });
    </script>
</body>
</html>

**Comment by NGPixel** [comment, marked as answer]
https://docs.requarks.io/en/troubleshooting#how-to-execute-javascript-code-on-page-load
